src/hotspot/cpu/aarch64/macroAssembler_aarch64.hpp

```diff
@@ -1380,6 +1380,27 @@ class MacroAssembler: public Assembler {
   void cache_wb(Address line);
   void cache_wbsync(bool is_pre);
 
+  // Code for java.lang.Thread::onSpinWait() intrinsic.
+  void spin_wait() {
+#define EMIT_N_INST(n, inst) for (int i = 0; i < (n); ++i) inst()
```

Comment:
Why use a macro here? You could just put the loop around the switch statement. And the method body seems sufficiently large that it ought to go in the .cpp file.

Reply:
Good point. There's no significant performance advantage to having this in the header.

Reply:
> Why use a macro here? You could just put the loop around the switch statement. And the method body seems sufficiently large that it ought to go in the .cpp file.

:) compiler engineering experience. Compilers have a problem to apply unswitching optimization to loop-invariant SWITCHes.
I'll update the code as suggested.

Reply:
Done